stl/inc/ranges

```diff
@@ -6837,6 +6837,664 @@ namespace ranges {
         inline constexpr _Stride_fn stride;
     } // namespace views
 
+#ifdef __cpp_lib_ranges_zip
+    namespace _Zip {
+        template <typename... _RangeTypes>
+        concept _Is_common = (sizeof...(_RangeTypes) == 1 && (_RANGES common_range<_RangeTypes> && ...)) //
+                          || (!(_RANGES bidirectional_range<_RangeTypes> && ...)
+                              && (_RANGES common_range<_RangeTypes> && ...)) //
+                          || ((_RANGES random_access_range<_RangeTypes> && ...)
+                              && (_RANGES sized_range<_RangeTypes> && ...));
+
+#ifndef __cpp_lib_tuple_like

```

Comment:
I want to get rid of `_Tuple_or_pair` right now. The mechanism makes the implementation and the (yet-to-write?) test difficult, and it will be eventually removed.

Reply:
You got it. I'll go ahead and just always use `std::tuple` in that case. Would that be acceptable, or would doing this block the PR from being accepted until the full P2165R4 gets implemented? If the latter is correct, then is there anything we could do to avoid having to wait?